### DISCUSSION TITLE
Query on "v6-only-preferred" option in DHCP Client

### DISCUSSION THREAD

**Discussion by arunkm00** [community/community#135128] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I have some queries about new dhcp option 108 i.e `v6-only-preferred`. Can someone clarify

1. Is there a way to inform dhclient that `IPv6Only `capability is enabled on the interface?
If not
- how does dhclient know when to send this option in DISCOVER message? Is it based on whether the option exist in `request `token from dhclient.conf file. Are we not checking whether the interface is capable or not?
- Say interface-1 needs option-108 and interface-2 does not need option-108. In this case, what is the recommendation?
Should I use different dhclient.conf file for each interface and differentiate in `request` token? 
or should I modify dhclient.conf file to have per interface `request` token? not sure whether it is possible.

2. Upon receiving option 108 from dhcp server, dhclient waits for V6ONLY_WAIT seconds. After the timer expiry, it goes back to the INIT state. What happens after that?
Does it repeat same process i.e send-discover, receive-offer, wait V6ONLY_WAIT time and so on? If so how long?
Does it stop after some time? or Does it get IPv4 too?

3. When both client and server support this option, is it not mandatory to stop IPv4 completely after client gets an IPv6 address ?

**Comment by LiteBrite82** [comment]
Hi @arunkm00 ,

Thanks for being a part of the GitHub Community, we're glad you're here!

It's possible that another GitHub user might have run into this same issue and can help, but the GitHub Community Discussions focuses primarily on topics related to GitHub itself or collaboration on project development and ideas. We want to make sure you’re getting the best support you can, but this space may not be the right place for this particular topic.

Best of luck!

**Reply by arunkm00** [reply]
Thanks @LiteBrite82.
The query is related to feature that is released in ISC-DHCP-Client 4.4.3 and code path is https://github.com/isc-projects/dhcp/
I am new to this git hub. I am really not sure where I should post this. Could you help here?
